biorxiv/polka_subset_experiment/05_publication_time_reviewer_request.py

```diff
@@ -0,0 +1,570 @@
+)
+polka_published_preprint_df.head()
+
+# +
+for col in ["preprint_date", "published_date"]:
+    polka_published_preprint_df[col] = pd.to_datetime(polka_published_preprint_df[col])
+
+polka_published_preprint_df["time_to_published"] = pd.to_timedelta(
+    polka_published_preprint_df["time_to_published"]
+)
+polka_published_preprint_df["days_to_published"] = polka_published_preprint_df[
+    "time_to_published"
+].dt.days
+print(polka_published_preprint_df.shape)
+polka_published_preprint_df.head()
+# -
+
+biorxiv_published_distances = pd.read_csv(
+    "../publication_delay_experiment/output/preprint_published_distances.tsv",
+    sep="\t",
+)
+biorxiv_published_distances["time_to_published"] = pd.to_timedelta(
+    biorxiv_published_distances["time_to_published"]
+)
+biorxiv_published_distances["days_to_published"] = biorxiv_published_distances[
+    "time_to_published"
+].dt.days
+biorxiv_published_distances = biorxiv_published_distances.query("days_to_published > 0")
+biorxiv_published_distances.head()
+
+# # Perform Regression Analysis - Version Count
+
+# ## Original
+
+# +
+# Get smoothed linear regression line
+x = biorxiv_published_distances.version_count.values.tolist()
+
+y = biorxiv_published_distances.time_to_published.apply(
+    lambda x: x / timedelta(days=1)
+).tolist()
+
+results_2 = linregress(x, y)
+print(results_2)
+# -
+
+x_line = np.array(
+    [
+        biorxiv_published_distances["version_count"].min(),
+        biorxiv_published_distances["version_count"].max(),
+    ]
+)
+y_line = x_line * results_2.slope + results_2.intercept
+
+# +
+# Get smoothed linear regression line
+polka_x = polka_published_preprint_df.version_count.values.tolist()
+
+polka_y = polka_published_preprint_df.time_to_published.apply(
+    lambda x: x / timedelta(days=1)
+).tolist()
+
+results_3 = linregress(polka_x, polka_y)
+print(results_3)
+# -
+
+polka_x_line = np.array(
+    [
+        polka_published_preprint_df["version_count"].min(),
+        polka_published_preprint_df["version_count"].max(),
+    ]
+)
+polka_y_line = polka_x_line * results_3.slope + results_3.intercept
+
+# +
+# Graph here?
+plt.figure(figsize=(11, 8.5))
+plt.rcParams.update({"font.size": 22})
+g = sns.violinplot(
+    x="version_count",
+    y="days_to_published",
+    data=biorxiv_published_distances,
+    cut=0,
+    scale="width",
+    palette="YlGnBu",
+)
+
+_ = g.set_ylabel("Time Elapsed Until Preprint is Published (Days)")
+_ = g.set_xlabel("# of Preprint Versions")
+_ = g.plot(x_line - 1, y_line, "--k")
+_ = g.plot(polka_x_line - 1, polka_y_line, "--k", color="red")
+_ = g.scatter(
+    polka_published_preprint_df["version_count"] - 1,
+    polka_published_preprint_df["days_to_published"],
+    c="red",
+    s=12,
+)
+_ = g.annotate(f"Y={results_2.slope:.2f}*X+{results_2.intercept:.2f}", (7, 1540))
+_ = g.annotate(
+    f"Y={results_3.slope:.2f}*X+{results_3.intercept:.2f}", (7, 1460), color="red"
```

Comment:
You may want to add a legend explaining the red regression line. I initially thought it was "expected time remaining once a publication gets to this version" but it's not that